addons/scopes/functions/fnc_getCurrentZeroRange.sqf

```diff
@@ -35,7 +35,11 @@ private _opticConfig = if (_optic != "") then {
 };
 
 private _zeroRange = currentZeroing _unit;
-if (GVAR(overwriteZeroRange) && {GVAR(canAdjustElevation) select _weaponIndex}) then {
+// Revert zeroing to default if overriding is enabled OR the selected sight's magnification is not higher than that of the naked eye, meaning that it is a secondary iron/holo sight
+if (
+    (GVAR(canAdjustElevation) select _weaponIndex) && 
+    {GVAR(overwriteZeroRange) || {getNumber (_opticConfig >> "ItemInfo" >> "OpticsModes" >> (_unit getOpticsMode _weaponIndex) >> "opticsZoomMax") > MIN_ZOOM_NAKEDEYE}}
```

Comment:
does `getOpticsMode` work on remote units? wiki isn't clear

Reply:
Good catch, could cause issues with rangecard copies.
Gonna be far from my PC for the next few days, should be able to test it thursday or friday.
Might be able to accept changes from mobile though.

Reply:
yeah, doesn't sync
checking `_p getOpticsMode 1, currentZeroing _p`
local: `ams, 100` or `iron, 200` depending on mode
remote: always just `ams, 100`

need to think some more but maybe something like (need to cache)
```

if (!local _unit) then {
    private _scope = (weaponsItems _unit) # _weaponIndex # 3;
    private _bestZoom = 1e6;
    private _assumedDistance = 99;
    {
        private _xZoom = getNumber (_x >> "opticsZoomMin");
        if (_xZoom < _bestZoom) then {
            _bestZoom = _xZoom;
            private _zeroing = getArray (_x >> "discreteDistance");
            // if ((count _zeroing) != 1) then { systemChat "?" };
            _assumedDistance = _zeroing param [0, 100];
        };
    } forEach configProperties [configFile >> "CfgWeapons" >> _scope >> "ItemInfo" >> "OpticsModes", "isClass _x"];
};
```

Reply:
i guess, if we assume the scope is the first optic mode, then it will probably work fine as is?

Reply:
Yes, I think that would suffice, the optic is already cached anyway for rangecard copy use, so either using its custom `ACE_Zeroing` or the ACE default zeroing setting should cover every scenario.

Reply:
Found the time to test this a bit on a DS with 2 client instances.
As far as I can tell, along the process of copying and opening a range card, the only code that has locality issues is in `fnc_getCurrentZeroRange.sqf`:
- `L30: private _optic = GVAR(Optics) select _weaponIndex;`, as it will return the optic of the user, not that of the copied unit. Easy fix with a ternary check for locality and alternate scope fetching.
- `L34: (configFile >> "CfgWeapons" >> (GVAR(Guns) select _weaponIndex))` returning the player's instead of the target unit's gun, again solvable with a simple check.
- `getOpticsMode` always returning the primary scope's config, which you identified.\
Assuming the fixes above are implemented, the first condition which sets `_zeroRange = GVAR(defaultZeroRange);` won't be entered if the player using the copy doesn't have an adjustable scope on his weapon.\
If both have an adjustable scope, `getOpticsMode` will always return the primary scope's zero, but since `currentZeroing` also always returns that value from a remote unit, it doesn't matter.\
I think the condition only needs an additional check for the unit being local, to save process time since it isn't required for remote units. The function's return value would then always either be `currentZeroing` of the main scope, or the custom `ACE_ScopeZeroRange` config value of the correct scope.

Reply:
Testing this with all recent changes, rangecard copy works well, I find no issues.

> It's called from both rangecards but more importantly the scope's firedEH on non-local units

This however is still a concern.
The fixes should resolve most issues the function could cause in the `firedEH` as well, with the exception of correctly adjusting trajectory for shots aimed with a secondary sight.
However, since those sights only have low magnification and would be used for close shots, the trajectory deviation should be negligible, no?

Reply:
I meant primary as in the "default" sight defined in the config, which appears to always be the magnified one as far as I've checked.
Yes, it's better than what we have.